lib/soundtrack.js

```diff
@@ -319,10 +320,58 @@ Soundtrack.prototype.trackFromSource = function(source, id, sourceCallback) {
   switch (source) {
     default:
       sourceCallback('Unknown source: ' + source);
-    break;
+    break;      
     case 'soundtrack':
       Track.findOne({ _id: id }).populate('_artist').exec( sourceCallback );
     break;
+    case 'bandcamp':
+      var obj = id;
+      Artist.findOne({ slug: slug(obj.artist) }).exec(function(err, artist) {
+        if (err || !artist) var artist = new Artist({ name: obj.artist });
+        
+        Track.findOne({
+          slug: slug(obj.title),
+          _artist: artist._id
+        }).exec(function(err, track) {
+          if (err || !track) var track = new Track({ title: obj.title, sources: obj.sources });
+          
+          track._artist = artist._id;
+          track.duration = obj.duration;
+
+          if (obj.thumbnail) {
+            track.images.thumbnail = obj.thumbnail;
+          }
+          track.sources.bandcamp = [ {
```

Comment:
It should really merge in the source type by track ID like the other source types do...